admin/data/sql/migrations/166.do.create-view-pupils-eligible-for-familiarisation-pin.sql

```diff
@@ -0,0 +1,45 @@
+--
+-- Find pupils eligible to generate a Familiarisation PIN
+--
+
+CREATE VIEW [mtc_admin].[vewPupilsEligibleForFamiliarisationPinGeneration] AS (
+        SELECT
+                p.id,
+                p.foreName,
+                p.middleNames,
+                p.lastName,
+                p.dateOfBirth,
+                p.school_id,
+                p.urlSlug
+         FROM
+                [mtc_admin].[pupil] p
+                LEFT JOIN [mtc_admin].[pupilAttendance] pa ON (p.id = pa.pupil_id)
+                LEFT JOIN [mtc_admin].[attendanceCode] ac ON (pa.attendanceCode_id = ac.id)
+                INNER JOIN [mtc_admin].[pupilStatus] ps ON (p.pupilStatus_id = ps.id)
+         WHERE
+                 -- include all pupils except those who are marked as not taking check because they left school
+                (ac.id IS NULL OR ac.code <> 'LEFTT')
+         AND    ps.code IN ('UNALLOC',
+                            'ALLOC',
+                            'LOGGED_IN',
+                            'NOT_TAKING_CHECK')
+
+
+         EXCEPT
```

Comment:
If we want to be able to generate a new familiarisation check _at_ _any_ _time_ e.g. because the access arrangements were changed, then we need to lose this `except` and add in a way to expire old familiarisation pins when generating a new familiarisation check.